staging/src/k8s.io/code-generator/cmd/conversion-gen/generators/conversion_test.go

```diff
@@ -0,0 +1,236 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_extractTag(t *testing.T) {
+	testCases := map[string]struct {
+		comments []string
+		expect   []string
+	}{
+		"no comments": {
+			comments: []string{},
+			expect:   nil,
+		},
+		"one comment": {
+			comments: []string{
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+			},
+			expect: []string{
+				"k8s.io/kubernetes/runtime.Object",
+			},
+		},
+		"two different comments": {
+			comments: []string{
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.List",
+			},
+			expect: []string{
+				"k8s.io/kubernetes/runtime.Object",
+				"k8s.io/kubernetes/runtime.List",
+			},
+		},
+		"two same comments": {
+			comments: []string{
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+			},
+			expect: []string{
+				"k8s.io/kubernetes/runtime.Object",
+				"k8s.io/kubernetes/runtime.Object",
+			},
+		},
```

Comment:
looking through the function, there are many more things to test. Non-tag lines for example. Or other kind of tags.

Reply:
yes, add other tags ut.